src/attached_probe.h

```diff
@@ -31,12 +31,28 @@ class AttachedProbe {
   const Probe &probe() const;
   int progfd() const;
   int linkfd_ = -1;
+  bool did_attach()
+  {
+    return attached_;
+  }
+  const std::string &err_msg() const
+  {
+    return err_msg_;
+  }
```

Comment:
This out of band status / error mechanism feels odd. How about unifying around `Result<T>` now that we're using it elsewhere? It'll take a little more refactoring, as the AttachedProbe constructor can't return anything, but that could easily be turned into `AttachedProbe() = delete;` and then have a static factory method that's something like `static Result<AttachedProbe> attach(...);`

`BPFtrace::attach_probe()` can return a vector of `Result<AttachedProbe>` and then still have `BPFtrace::run()` do all the error handling.

WDYT?

Reply:
Sounds like a bit of a pain TBH 😄 but I agree that it's a better pattern. Mind if I look at doing that refactoring in a follow-up PR? I think it'll be a cleaner diff.

Reply:
@danobi ok, it wasn't too bad. I made this change, let me know if it's what you had in mind.